tests/CMakeLists.txt

```diff
@@ -0,0 +1,21 @@
+FIND_PACKAGE(Threads REQUIRED)
+
+SET(Boost_USE_MULTITHREADED ON)  
+FIND_PACKAGE(Boost REQUIRED COMPONENTS unit_test_framework REQUIRED)
+ADD_DEFINITIONS("-DBOOST_TEST_DYN_LINK")
+
+ADD_LIBRARY(vanilla_all OBJECT
+        vanilla_all.cpp)
+
+ADD_EXECUTABLE(vanilla_tests
+        $<TARGET_OBJECTS:vanilla_all>
+        main.cpp)
+
+TARGET_LINK_LIBRARIES(vanilla_tests
```

Comment:
`main.cpp` only needs the unit_test_framework, others’ dependencies come cometransitively

Reply:
I think @venediktov  is right. Adding ${Boost_LIBRARIES}  will solve the problem. 
By adding more and more test we would need all required boost libraries anyway.

Reply:
from official boost document we may want to use  [${Boost_UNIT_TEST_FRAMEWORK_LIBRARY}](https://www.boost.org/doc/libs/1_62_0/libs/test/doc/html/boost_test/section_faq.html)
```
cmake_minimum_required(VERSION 2.8.11)
project(my_first_test)
enable_testing()

# replace XX with the version you have
set(Boost_ADDITIONAL_VERSIONS "1.XX" "1.XX.0")

# finds boost, triggers an error otherwise
find_package(Boost XX REQUIRED COMPONENTS unit_test_framework)

# creates the executable
add_executable(test_executable test_file.cpp)
# indicates the include paths
target_include_directories(test_executable PRIVATE ${Boost_INCLUDE_DIRS})
# indicates the shared library variant
target_compile_definitions(test_executable PRIVATE "BOOST_TEST_DYN_LINK=1")
# indicates the link paths
target_link_libraries(test_executable ${Boost_UNIT_TEST_FRAMEWORK_LIBRARY})

# declares a test with our executable
add_test(NAME test1 COMMAND test_executable)
```